subhub/tests/conftest.py

```diff
@@ -1,12 +1,52 @@
+import logging
+import os
+import signal
+import subprocess
+
+import psutil
 import pytest
 from flask import g
 
 from subhub.api import payments
 from subhub.app import create_app
 
+ddb_process = None
+
+
+def pytest_configure():
+    """Called before testing begins"""
+    global ddb_process
+    for name in ('boto3', 'botocore', 'stripe'):
+        logging.getLogger(name).setLevel(logging.CRITICAL)
+    if os.getenv("AWS_LOCAL_DYNAMODB") is None:
+        os.environ["AWS_LOCAL_DYNAMODB"] = "http://127.0.0.1:8000"
+
+    # Latest boto3 now wants fake credentials around, so here we are.
+    os.environ["AWS_ACCESS_KEY_ID"] = "fake"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "fake"
+
+    cmd = " ".join(["./services/fxa/node_modules/.bin/dynalite --port 8000"])
+    ddb_process = subprocess.Popen(cmd, shell=True, env=os.environ,
+                                   stdout=subprocess.PIPE)
+    while 1:
+        line = ddb_process.stdout.readline()
+        if line.startswith(b"Listening"):
+            break
+
+
+def pytest_unconfigure():
+    global ddb_process
+    """Called after all tests run and warnings displayed"""
+    proc = psutil.Process(pid=ddb_process.pid)
+    child_procs = proc.children(recursive=True)
+    for p in [proc] + child_procs:
+        os.kill(p.pid, signal.SIGTERM)
+    ddb_process.wait()
+
 
 @pytest.fixture(scope="module")
 def app():
+    print(f'test app')
```

Comment:
'f' for formatting not required for this string as there is no {} interpolation

Reply:
agree